docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx

```diff
@@ -0,0 +1,259 @@
+---
+title: How to fetch and stream data in your application
+nav_title: Fetching Data and Streaming
+description: Learn how to fetch and stream data in your Next.js application.
+related:
+  links:
+    - app/api-reference/functions/fetch
+    - app/api-reference/file-conventions/loading
+---
+
+There are a few ways to fetch data in your Next.js application. This page will walk you through how you can fetch data in [Server Components](#server-components) and [Client Components](#client-components). As well as how to to [stream](#streaming) content that depends on data.
+
+## Fetching Data
+
+### Server Components
+
+You can fetch data in Server Components using:
+
+1. The [`fetch` API](#with-the-fetch-api)
+2. An [ORM or database](#with-an-orm-or-database)
+
+#### With the `fetch` API
+
+To fetch data with the `fetch` API, turn your component into an async function, and await the `fetch` call. For example:
+
+```tsx filename="app/blog/page.tsx" switcher
+export default async function Page() {
+  const data = await fetch('https://api.vercel.app/blog')
+  const posts = await data.json()
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+```jsx filename="app/blog/page.js" switcher
+export default async function Page() {
+  const data = await fetch('https://api.vercel.app/blog')
+  const posts = await data.json()
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+#### With an ORM or database
+
+You can fetch data with an ORM or database by turning your component into an `async` function, and awaiting the call:
+
+```tsx filename="app/blog/page.tsx" switcher
+import { db, posts } from '@/lib/db'
+
+export default async function Page() {
+  const allPosts = await db.select().from(posts)
+  return (
+    <ul>
+      {allPosts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+```jsx filename="app/blog/page.js" switcher
+import { db, posts } from '@/lib/db'
+
+export default async function Page() {
+  const allPosts = await db.select().from(posts)
+  return (
+    <ul>
+      {allPosts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+### Client Components
+
+To fetch data in Client Components, we recommend using a React library like [SWR](https://swr.vercel.app/) or [React Query](https://tanstack.com/query/latest). For example:
+
+```tsx filename="app/blog/page.tsx" switcher
+'use client'
+
+import useSWR from 'swr'
+import { fetcher } from '@/lib/fetcher'
```

Comment:
Can we give a small example of what this `fetcher` function is/does?

Reply:
Yes, can do! It's something you set up with SWR.